virtcontainers/clh.go

```diff
@@ -877,45 +856,61 @@ func (clh *cloudHypervisor) getAvailableVersion() error {
 
 func (clh *cloudHypervisor) LaunchClh() (string, int, error) {
 
-	errStr := ""
-
 	clhPath, err := clh.clhPath()
 	if err != nil {
 		return "", -1, err
 	}
 
 	args := []string{cscAPIsocket, clh.state.apiSocket}
 	if clh.config.Debug {
-
-		logfile, err := clh.logFilePath(clh.id)
-		if err != nil {
-			return "", -1, err
-		}
-		args = append(args, cscLogFile)
-		args = append(args, logfile)
+		args = append(args, "-vvv")
 	}
 
 	clh.Logger().WithField("path", clhPath).Info()
 	clh.Logger().WithField("args", strings.Join(args, " ")).Info()
 
-	cmd := exec.Command(clhPath, args...)
-	cmd.Stdout = &clh.cmdOutput
-	cmd.Stderr = &clh.cmdOutput
+	cmdHypervisor := exec.Command(clhPath, args...)
 
 	if clh.config.Debug {
-		cmd.Env = os.Environ()
-		cmd.Env = append(cmd.Env, "RUST_BACKTRACE=full")
+		cmdHypervisor.Env = os.Environ()
+		cmdHypervisor.Env = append(cmdHypervisor.Env, "RUST_BACKTRACE=full")
+	}
+
+	hypervisoOutput, err := cmdHypervisor.StdoutPipe()
+	if err != nil {
+		return "", -1, err
 	}
+	cmdHypervisor.Stderr = cmdHypervisor.Stdout
```

Comment:
Also a question: is it safe to open a pipe with `stdoutPipe()` but not read from it (when debug is off)?

It looks like we are ignoring the stdout/stderr of CLH anyway when debug is off. Will it be the same if we open the pipe and redirect `cmdHypervisor.Stdour/Stderr` to it only when debug is on ( and do noting to the stderr/out when debug is off)?

Reply:
It is a good question, I think the system has a buffer where it dumps data, in the case when is ignore I imagine new prints to stdout, stderr will get a try again error, not sure how rust handle this. @sboeuf any suggestion here?